experimental/dds/tree2/src/shared-tree/sharedTreeChangeCodecs.ts

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IJsonCodec, ICodecOptions, ICodecFamily, makeCodecFamily } from "../codec";
+import { Mutable } from "../util";
+import {
+	ModularChangeset,
+	makeModularChangeCodec,
+	makeSchemaChangeCodec,
+} from "../feature-libraries";
+import { SharedTreeChange } from "./sharedTreeChangeTypes";
+
+// These can't be an interfaces or they don't get the special string indexer bonus property.
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+type EncodedModularChange = {
+	type: "data";
```

Comment:
Abram used \src\codec\discriminatedUnions.ts for representing discriminated enums in the wire format for `MarkEffect`. You may want to check with him if it makes sense to do the same here.